### PR TITLE
Allow hardcoded ID passed in scheduling a task

### PR DESCRIPTION
**Pull request by tsullivan**
Allows schedulers to upsert their tasks

https://github.com/elastic/kibana/pull/23555#pullrequestreview-159991409

### CODE REVIEW

**Comment by tsullivan** [comment]
Probably want this to add a test in test/plugin_functional/plugins/sample_task_plugin/init_routes.js that ensures passing `id` works as intended

File: src/server/task_manager/task_store.ts [hunk 1]
```diff
@@ -162,10 +162,10 @@ export class TaskStore {
       throw new Error(`Unsupported task type "${taskInstance.taskType}".`);
     }
 
-    const body = rawSource(taskInstance);
-    const { task } = body;
+    const { id, type, task } = rawSource(taskInstance);
```
Comment (chrisdavies) [inline src/server/task_manager/task_store.ts]: If we ever change raw source to return a new property, we'll have to modify this to structure the new property. Maybe this should be `{ id, ...body}` or something?

**Review by chrisdavies — APPROVED** [review]
LGTM w/ a tweak, maybe.

**Review by chrisdavies — APPROVED** [review]
LGTM!

**Pull request merged**